### PR TITLE
Python: Azure AI Inference connector code quality improvement

### PR DESCRIPTION
**Pull request by TaoChenOSU**
### Motivation and Context

We have been working on improving code quality and increasing the test coverage. This PR focuses on the Azure AI Inference connector.

### Description

1. Improve test coverage.
image
2. Cleared all Mypy warnings.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py [hunk 1]
```diff
@@ -124,6 +127,9 @@ async def get_chat_message_contents(
         Returns:
             A list of chat message contents.
         """
+        settings = self.get_prompt_execution_settings_from_settings(settings)
```
Comment (moonbox3) [inline python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py:130]: Is there a need to always call this even if the settings are already of the expected `AzureAIInferenceChatPromptExecutionSettings` type?

**Review by moonbox3 — APPROVED** [review]

File: python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py (hunk 1 above)
Comment (TaoChenOSU) [reply python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py:130]: This method checks if the settings is of the right type. If so, it just returns the settings untouched.

File: python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py [hunk 2]
```diff
@@ -408,9 +416,11 @@ def _verify_function_choice_behavior(
                 )
 
     def _configure_function_choice_behavior(
-        self, settings: AzureAIInferenceChatPromptExecutionSettings, kernel: Kernel
+        self, settings: AzureAIInferenceChatPromptExecutionSettings, kernel: Kernel | None
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py:419]: I don't like this change, most likely people will not call this directly, but this combined with an assert means inconsistent behavior. Prob be better to check if `kernel` is not None inside the auto invoke specific version

File: python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py [hunk 3]
```diff
@@ -419,13 +429,15 @@ async def _invoke_function_calls(
         self,
         function_calls: list[FunctionCallContent],
         chat_history: ChatHistory,
-        kernel: Kernel,
+        kernel: Kernel | None,
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py:432]: Then this one can be Kernel as well

File: python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_text_embedding.py [hunk 4]
```diff
@@ -73,9 +74,20 @@ def __init__(
...
+        **kwargs: Any,
+    ) -> ndarray:
         """Generate embeddings from the Azure AI Inference service."""
-        settings: AzureAIInferenceEmbeddingPromptExecutionSettings = kwargs.get("settings", None)
+        if not settings:
+            settings = AzureAIInferenceEmbeddingPromptExecutionSettings()
+        else:
+            settings = self.get_prompt_execution_settings_from_settings(settings)
+        assert isinstance(settings, AzureAIInferenceEmbeddingPromptExecutionSettings)  # nosec
+        assert isinstance(self.client, EmbeddingsClient)  # nosec
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_text_embedding.py:89]: Would it be possible to set client in this class to the right type?

File: python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py (hunk 2 above)
Comment (TaoChenOSU) [reply python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py:419]: This is primarily for Mypy. The way we get the kernel from the keyword arguments means the kernel could potentially be None.

File: python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py (hunk 3 above)
Comment (TaoChenOSU) [reply python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py:432]: Same as above

File: python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_text_embedding.py (hunk 4 above)
Comment (TaoChenOSU) [reply python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_text_embedding.py:89]: The client is a union type but Mypy cannot narrow it down.

**Review by eavanvalkenburg — APPROVED** [review]

**Pull request merged**